### PR TITLE
Update score and rank of all teams of the contest after rejudging.

### LINKED ISSUE #2105: First to solve rejudging
### Description of the problem
There can be multiple persons with "first to solve" on the scoreboard due to rejudgings

### Your environment
DOMjudge 8.3.0DEV/5d0968d34

### Steps to reproduce
- have two submissions by two users (first gets WA later gets AC)
- (change a testcase for example because it was invalid)
- rejudge both submissions (now both AC)

### Expected behaviour
Only the first person gets the "first to solve" on the scoreboard after the rejudging

### Actual behaviour
Both get the "first to solve" on the scoreboard after the rejudging

### PR DESCRIPTION
**Pull request by nickygerritsen**
This makes sure the rank and first to solve of all teams are correctly set.
Fixes #2105.

### CODE REVIEW

File: webapp/tests/Unit/Service/RejudgingServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        $entityManager->persist($newJudging);
+        $entityManager->flush();
+
+        // Now apply the rejudging
+        $rejudgingService->finishRejudging($rejudging, RejudgingService::ACTION_APPLY);
+
+        // Finally, get the scoreboard again and test if the first to solve changed
+        $scoreboard = $scoreboardService->getScoreboard($contest, true);
+
+        static::assertFalse($scoreboard->solvedFirst($team1, $contestProblem));
```
Comment (vmcj) [inline webapp/tests/Unit/Service/RejudgingServiceTest.php:78]: Before this failed right? We would have 2 teams with `solvedFirst`.

**Review by vmcj — COMMENTED** [review]
Based on the test I would say this is correct. But I think someone else should take a look and give approval.

**Review by meisterT — COMMENTED** [review]
Can we easily identify which teams to update (the teams who currently are first to solve for these problems) instead of all teams in a contest?

**Comment by nickygerritsen** [comment]
But the rank of teams can also change right?

**Comment by meisterT** [comment]
True. How did we never run into this issue then?

**Comment by nickygerritsen** [comment]
> True. How did we never run into this issue then?

🤷‍♂️

**Review by meisterT — APPROVED** [review]

**Pull request merged**